index.html

```diff
@@ -902,32 +902,64 @@
-preferences, shopping carts, etc. The web platform offers many mechanisms that
-are either intended to accomplish this recognition or that can be trivially used
-for it, including [[RFC6265|cookies]], {{WindowLocalStorage/localStorage}},
-{{WindowOrWorkerGlobalScope/indexedDB}}, {{CacheStorage}}, and other forms of
-storage.
+As described in [[[#identity]]], [=cross-context recognition=] can sometimes be [=appropriate=], but
+users need to be able to control when websites do it as much as possible.
 
-A privacy harm only occurs if the user wants to break the association between
-two visits, but the site can still determine with high probability that the two
-visits came from the same user.
+<div class="practice">
 
-A user might expect that their two visits won't be associated if they:
+<p><span class="practicelab" id="principle-prevent-cross-partition-recognition">[=User agents=]
+should ensure that, if a [=person=] visits two or more web pages from different [=partitions=], that
+the pages cannot quickly determine that the visits probably came from the same person, for any
+significant or involuntary fraction of the people who use the web, unless the person explicitly
+expresses the same [=identity=] to the visits, or preventing this correlation would break a
+technical feature that is fundamental to the Web.</span></p>
 
-* Use a browser that promises to avoid such correlation.
-* Use their browser's private browsing mode. ([[WHAT-DOES-PRIVATE-BROWSING-DO]])
-* Use two different browser profiles between the two visits.
-* Explicitly clear the site's cookies or storage.
+</div>
 
-This recognition is generally accomplished by either "supercookies" or <a
-data-cite="fingerprinting-guidance#dfn-browser-fingerprinting">browser
+* This principle uses "probably" because websites can do harm even if they can't be completely
+  certain that visits come from the same person.
+* This principle uses "quickly" because it currently appears impossible to prevent some forms of
+  fingerprinting that take a long time or many visits within each [=partition=].
+* This principle allows violations that only affect a small fraction of people who use the web
+  because people need to be allowed to configure their systems in unique ways, for example by using
+  a browser with a very small number of users. As long as a tracker can't track a significant number
+  of people, it's likely to be unviable to maintain the tracker. However, this doesn't excuse making
+  small groups of people trackable when those people didn't choose to be in the group.
+* This principle also allows violations when preventing those violations would break fundamental
+  aspects of the web. In many cases it's possible to change the design in a way that avoids the
+  violation without breaking valid use cases, but for cases where that's not possible this document
+  delegates to other documents, for example the [[[PRIVACY-THREAT]]], to discuss what detailed
+  tradeoffs to make.
```

Comment:
How's this @sandandsnow?

Reply:
@jyasskin, thank you for this revision. I think this works, though I still would prefer if we could find a way to avoid saying "the principle allows violations ..." in the document and instead say something along the lines of "in very limited circumstances, such as xyz, it may be impossible to guarantee that the effect of this principle applies to all users". (That wording probably is not right either, but I would prefer to move away from "allowing violations".